### ISSUE TITLE
Creating a folder doesn't allow me to set the `uid` like the API

### ISSUE THREAD

**Issue by alexross1988IBM** [grafana/terraform-provider-grafana#241]
The API allows this: https://grafana.com/docs/grafana/latest/http_api/folder/#create-folder
The provider does not https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/folder

I've tried for certainty too:
```
╷
│ Error: Value for unconfigurable attribute
│ 
│   with grafana_folder.collection,
│   on metis.tf line 52, in resource "grafana_folder" "collection":
│   52:   uid = local.ENVIRONMENT_NAME
│ 
│ Can't configure a value for "uid": its value will be decided automatically
│ based on the result of applying this configuration.
╵
```

Is this an enhancement that can be added?
If a PR came from ourselves would you be willing to accept it? (Not committing to this, just exploring our options)

**Comment by feniix** [comment]
Is there a technical reason for this behaviour in the resource?

**Comment by alexross1988IBM** [comment]
We use it to keep control of direct URLs to dashboards

**Comment by feniix** [comment]
yes, we do too, I meant to ask the grafana developers if there is a technical reason why the api or the datasource does not support it

**Comment by julienduchesne** [comment]
Support for this was released in [v1.21.0](https://github.com/grafana/terraform-provider-grafana/releases/tag/v1.21.0)
